compiler/src/iree/compiler/Dialect/LinalgExt/IR/AggregatedOpInterfaceImpl.cpp

```diff
@@ -910,77 +920,61 @@ FailureOr<SmallVector<Value>> Im2colOp::decomposeOperation(OpBuilder &b) {
   int64_t inputRank = getInputRank();
   int64_t outputRank = getOutputRank();
 
-  // For now, only extract a 1D slice when the batch dimension corresponds to a
-  // contiguous slice and produces a rank reduced/expanded result.
-  // TODO: check if 1d slice extraction can be used for all cases and thus
-  // simplify the codes.
-  SmallVector<OpFoldResult> inputTileSizes;
-  if (contiguousAlongB && inputRank != outputRank) {
-    inputTileSizes.push_back(innerInputTileSize);
-  } else {
-    int64_t minRank = std::min<int64_t>(inputRank, outputRank);
-    for (int i = 0; i < minRank - 1; i++) {
-      inputTileSizes.push_back(b.getIndexAttr(1));
+  // For now, only extract a 1D slice when the vectorized dim is not innermost
+  // in the output, and the input and output ranks are different. Otherwise,
+  // try to preserve the original rank to avoid rank reducing slices.
+  int64_t sliceRank = std::min<int64_t>(inputRank, outputRank);
```

Comment:
It's generally discouraged to pass template arguments to `min`/`max` -- cast the arguments instead